VotingApplication/VotingApplication.Web/Api/Controllers/PollResultsController.cs

```diff
@@ -105,31 +84,46 @@ private static DateTime UnixTimeToDateTime(double unixTimestamp)
             return dateTime;
         }
 
-        private static ResultsRequestResponseModel GenerateResults(IEnumerable<Vote> votes, bool namedVoting)
+        private static ResultsRequestResponseModel GenerateResults(IEnumerable<Vote> votes, IEnumerable<Choice> choices, bool namedVoting)
         {
-            var groupedResults = (from vote in votes
-                                  group vote by vote.Choice.PollChoiceNumber into result
-                                  let optionGroupVotes = result.ToList()
-                                  select new
-                                             {
-                                                 ChoiceName = optionGroupVotes[0].Choice.Name,
-                                                 Sum = optionGroupVotes.Sum(v => v.VoteValue),
-                                                 Voters = optionGroupVotes.Select(v => CreateResultVoteModel(v, namedVoting)).ToList()
-                                             })
-                  .ToList();
-
-            if (!groupedResults.Any())
+            if (choices.Count() == 0)
```

Comment:
`.Count` enumerates the whole collection. To check if there's anything in the collection, `.Any()` terminates as soon as one is found, and should be used instead of `.Count() == 0`.